run_tests.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# Script to run tests and benchmarks.
+
+import glob
+import os
+import subprocess
+
+
+def _convert_path_to_module(test_script : str) -> str:
+  """Convert the path of the test script to its module name."""
+  test_script = test_script.replace(os.sep, ".")
+  test_script = test_script.strip(".")
+  if test_script.endswith(".py"):
+    return test_script[:-3]
+  return test_script
+
+
+def _run_test(test_script: str) -> bool:
+  """Run the provided test script an return failure or success.
+
+  A test succeeds if:
+  - it does not time out
+  - it returns zero
+  - it does not print FAILURE
+  """
+  print(f"- running {test_script}: ", end="")
+  module = _convert_path_to_module(test_script)
+  proc = subprocess.Popen(["python", "-m", module],
+                          stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  try:
+      outs, errs = proc.communicate(timeout=20)
+  except subprocess.TimeoutExpired:
+      proc.kill()
+      print("\033[31m" + "FAILED" + "\033[m")
+      print("  -> test execution timed out")
+      return False
+  if proc.returncode != 0:
+      print("\033[31m" + "FAILED" + "\033[m")
+      print(f"  -> test returned code {proc.returncode}")
```

Comment:
you prob want to return this somehow so that the CI can pick it up

Reply:
I made the script return non zero if there was a failure and did add some github action. I doubt it will work out of the box though.